### PR TITLE
Creating a shared Project Update Manager

### PR DESCRIPTION
**Pull request by lancewf**
### :nut_and_bolt: Description

Merging the compliance and ingest project update manager into one shared manager in the lib directory. This simplifies updating and troubleshooting problems in the two managers.

### :athletic_shoe: Demo Script / Repro Steps

Same demo script from this PR https://github.com/chef/automate/pull/132

### :chains: Related Resources

refactors this PR - https://github.com/chef/automate/pull/132
https://github.com/chef/automate/issues/62

### :white_check_mark: Checklist

- [x] Necessary tests added/updated?
- [ ] Necessary docs added/updated?
- [x] Code actually executed?
- [x] Vetting performed (unit tests, lint, etc.)?

### CODE REVIEW

File: .bldr.toml [hunk 1]
```diff
@@ -210,11 +210,13 @@ paths = [
   "components/automate-grpc/*",
   "components/compliance-service/*",
   "components/config-mgmt-service/*",
+  "components/event-service/*",
```
Comment (lancewf) [inline .bldr.toml]: Had to run `generate_bldr_config`

File: components/compliance-service/config/config_mgmt.go [hunk 2]
```diff
@@ -4,27 +4,16 @@ import (
 	"io/ioutil"
 	"os"
 
+	project_update_lib "github.com/chef/automate/lib/authz"
```
Comment (lancewf) [inline components/compliance-service/config/config_mgmt.go]: Updating to use the common config struct.

File: components/compliance-service/ingest/ingestic/ingestic.go [hunk 3]
```diff
@@ -19,12 +20,6 @@ type ESClient struct {
 	initialized bool
 }
 
-type JobStatus struct {
```
Comment (lancewf) [inline components/compliance-service/ingest/ingestic/ingestic.go]: This was moved to the common lib directory

File: components/compliance-service/ingest/ingestic/ingestic.go [hunk 4]
```diff
@@ -198,6 +193,25 @@ func (backend *ESClient) setDailyLatestToFalse(ctx context.Context, nodeId strin
 	return err
 }
 
+func (backend *ESClient) UpdateProjectTags(ctx context.Context, projectTaggingRules map[string]*iam_v2.ProjectRules) ([]string, error) {
```
Comment (lancewf) [inline components/compliance-service/ingest/ingestic/ingestic.go]: This is the function that the common project update manager calls to start the update process.

File: components/compliance-service/ingest/server/compliance.go [hunk 5]
```diff
@@ -38,7 +39,7 @@ type ComplianceIngestServer struct {
 	mgrClient          manager.NodeManagerServiceClient
 	automateURL        string
 	notifierClient     notifier.Notifier
-	updateManager      *projectupdater.Manager
+	updateManager      *project_update_lib.DomainProjectUpdateManager
```
Comment (lancewf) [inline components/compliance-service/ingest/server/compliance.go]: Removed the specific compliance project update manager to use the common domain project update manager

File: components/ingest-service/backend/elastic/elastic.go [hunk 6]
```diff
@@ -38,17 +42,36 @@ func (es *Backend) Initializing() bool {
...
-func (es *Backend) JobStatus(ctx context.Context, jobID string) (backend.JobStatus, error) {
+func (es *Backend) UpdateProjectTags(ctx context.Context, projectTaggingRules map[string]*iam_v2.ProjectRules) ([]string, error) {
+	logrus.Debug("starting project updater")
+
+	esNodeJobID, err := es.UpdateNodeProjectTags(ctx, projectTaggingRules)
+	if err != nil {
+		return []string{}, errors.Wrap(err, "Failed to start Elasticsearch Node project tags update")
+	}
+
+	esActionJobID, err := es.UpdateActionProjectTags(ctx, projectTaggingRules)
```
Comment (lancewf) [inline components/ingest-service/backend/elastic/elastic.go]: This adds the actions to the update manager process. https://github.com/chef/automate/issues/62

File: lib/authz/project_update_manager.go [hunk 7]
```diff
@@ -1,23 +1,18 @@
-package projectupdater
+package authz
```
Comment (lancewf) [inline lib/authz/project_update_manager.go]: This is the main change. There are two new interfaces EsClient and ConfigManager. The EsClient allows the start, stop, and status of the elasticsearch jobs. The ConfigManager allows the retrieving and storing of the stage status.

File: components/compliance-service/config/config_mgmt.go (hunk 2 above)
Comment (phiggins) [reply components/compliance-service/config/config_mgmt.go]: nitpick: Any reason to not just leave this as `authz`?

Comment (lancewf) [reply components/compliance-service/config/config_mgmt.go]: I only thought that the `project_update_lib` was a better description of the package.

**Pull request merged**